lambda_functions/analyzer/binary_info.py

```diff
@@ -43,23 +44,35 @@ def __str__(self) -> str:
         return self.s3_identifier
 
     def _download_from_s3(self) -> None:
-        """Download binary from S3 and measure elapsed time."""
-        LOGGER.debug('Downloading %s to %s', self.object_key, self.download_path)
+        """Download binary from S3 and measure elapsed time, if it's under the size limit."""
+        file_size = analyzer_aws_lib.get_file_size_from_s3(self.bucket_name, self.object_key)
+
+        if file_size > self.MAX_FILE_SIZE_BYTES:
+            LOGGER.warning(f'File {self.object_key} is too large ({file_size} bytes). Skipping download.')
+            return  # Skip the download
 
+        LOGGER.debug('Downloading %s to %s', self.object_key, self.download_path)
         start_time = time.time()
         self.s3_last_modified, self.s3_metadata = analyzer_aws_lib.download_from_s3(
             self.bucket_name, self.object_key, self.download_path)
         self.download_time_ms = (time.time() - start_time) * 1000
 
-    def __enter__(self) -> Any:  # mypy/typing doesn't support recursive type yet
+    def __enter__(self) -> Any:
         """Download the binary from S3 and run YARA analysis."""
+
         self._download_from_s3()
-        self.computed_sha, self.computed_md5 = file_hash.compute_hashes(self.download_path)
 
-        LOGGER.debug('Running YARA analysis')
-        self.yara_matches = self.yara_analyzer.analyze(
-            self.download_path, original_target_path=self.filepath
-        )
+        # Only proceed with analysis if the file was actually downloaded
+        if os.path.exists(self.download_path):
```

Comment:
The necessary information (file downloaded / skipped) should be returned from `_download_from_s3` in my opinion. This would the code here to check whether to proceed without having to look at the file system.